### ISSUE TITLE
system > Reids Managment says "No date part in '' found"

### ISSUE THREAD

**Issue by nilands55s** [steverobbins/Magento-Redismanager#20]
I am new to redis and I set it up with colinmollenhour/Cm_RedisSession, but I also have other Magento redis files that I don't think I am using. colinmollenhour/Cm_RedisSession was setup with modan. I am hoping it is working right, but the magento backend slows down after many hours and works faster when I flush it via SSH. This however is a pain in the! The error page has this:

```
No date part in '' found.

Trace:
#0 /home/niland5/public_html/lib/Zend/Date.php(1078): Zend_Date->_calculate('set', false, NULL, 'en_US')
#1 /home/niland5/public_html/lib/Zend/Date.php(197): Zend_Date->set(false, NULL, 'en_US')
#2 /home/niland5/public_html/app/code/core/Mage/Core/Model/Locale.php(478): Zend_Date->__construct(false, NULL, Object(Zend_Locale))
#3 /home/niland5/public_html/app/code/core/Mage/Core/Helper/Data.php(187): Mage_Core_Model_Locale->date(false)
#4 /home/niland5/public_html/app/code/community/Steverobbins/Redismanager/Block/Adminhtml/Manager.php(166): Mage_Core_Helper_Data->formatTime(1449955285, 'long')
#5 /home/niland5/public_html/app/code/community/Steverobbins/Redismanager/Block/Adminhtml/Manager.php(92): Steverobbins_Redismanager_Block_Adminhtml_Manager->_getLastSave()
#6 /home/niland5/public_html/app/code/community/Steverobbins/Redismanager/Block/Adminhtml/Manager.php(55): Steverobbins_Redismanager_Block_Adminhtml_Manager->_getSortedService(Array, 0, Object(Steverobbins_Redismanager_Model_Backend_Redis_Mage))
#7 /home/niland5/public_html/app/design/adminhtml/default/default/template/steverobbins/redismanager/grid.phtml(14): Steverobbins_Redismanager_Block_Adminhtml_Manager->getSortedServices()
#8 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(241): include('/home/niland5/p...')
#9 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(272): Mage_Core_Block_Template->fetchView('adminhtml/defau...')
#10 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(286): Mage_Core_Block_Template->renderView()
#11 /home/niland5/public_html/app/code/core/Mage/Adminhtml/Block/Template.php(81): Mage_Core_Block_Template->_toHtml()
#12 /home/niland5/public_html/app/code/core/Mage/Core/Block/Abstract.php(863): Mage_Adminhtml_Block_Template->_toHtml()
#13 /home/niland5/public_html/app/code/core/Mage/Core/Block/Abstract.php(582): Mage_Core_Block_Abstract->toHtml()
#14 /home/niland5/public_html/app/code/core/Mage/Core/Block/Abstract.php(526): Mage_Core_Block_Abstract->_getChildHtml('redismanager.gr...', true)
#15 /home/niland5/public_html/app/design/adminhtml/default/default/template/steverobbins/redismanager/manager.phtml(26): Mage_Core_Block_Abstract->getChildHtml('redismanager.gr...')
#16 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(241): include('/home/niland5/p...')
#17 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(272): Mage_Core_Block_Template->fetchView('adminhtml/defau...')
#18 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(286): Mage_Core_Block_Template->renderView()
#19 /home/niland5/public_html/app/code/core/Mage/Adminhtml/Block/Template.php(81): Mage_Core_Block_Template->_toHtml()
#20 /home/niland5/public_html/app/code/core/Mage/Core/Block/Abstract.php(863): Mage_Adminhtml_Block_Template->_toHtml()
#21 /home/niland5/public_html/app/code/core/Mage/Core/Block/Text/List.php(43): Mage_Core_Block_Abstract->toHtml()
#22 /home/niland5/public_html/app/code/core/Mage/Core/Block/Abstract.php(863): Mage_Core_Block_Text_List->_toHtml()
#23 /home/niland5/public_html/app/code/core/Mage/Core/Block/Abstract.php(582): Mage_Core_Block_Abstract->toHtml()
#24 /home/niland5/public_html/app/code/core/Mage/Core/Block/Abstract.php(526): Mage_Core_Block_Abstract->_getChildHtml('content', true)
#25 /home/niland5/public_html/app/design/adminhtml/default/default/template/page.phtml(74): Mage_Core_Block_Abstract->getChildHtml('content')
#26 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(241): include('/home/niland5/p...')
#27 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(272): Mage_Core_Block_Template->fetchView('adminhtml/defau...')
#28 /home/niland5/public_html/app/code/core/Mage/Core/Block/Template.php(286): Mage_Core_Block_Template->renderView()
#29 /home/niland5/public_html/app/code/core/Mage/Adminhtml/Block/Template.php(81): Mage_Core_Block_Template->_toHtml()
#30 /home/niland5/public_html/app/code/core/Mage/Core/Block/Abstract.php(863): Mage_Adminhtml_Block_Template->_toHtml()
#31 /home/niland5/public_html/app/code/core/Mage/Core/Model/Layout.php(529): Mage_Core_Block_Abstract->toHtml()
#32 /home/niland5/public_html/app/code/core/Mage/Core/Controller/Varien/Action.php(391): Mage_Core_Model_Layout->getOutput()
#33 /home/niland5/public_html/app/code/community/Steverobbins/Redismanager/controllers/Adminhtml/RedismanagerController.php(56): Mage_Core_Controller_Varien_Action->renderLayout()
#34 /home/niland5/public_html/app/code/core/Mage/Core/Controller/Varien/Action.php(420): Steverobbins_Redismanager_Adminhtml_RedismanagerController->indexAction()
#35 /home/niland5/public_html/app/code/core/Mage/Core/Controller/Varien/Router/Standard.php(254): Mage_Core_Controller_Varien_Action->dispatch('index')
#36 /home/niland5/public_html/app/code/core/Mage/Core/Controller/Varien/Front.php(176): Mage_Core_Controller_Varien_Router_Standard->match(Object(Mage_Core_Controller_Request_Http))
#37 /home/niland5/public_html/app/code/local/Mage/Core/Model/App.php(349): Mage_Core_Controller_Varien_Front->dispatch()
#38 /home/niland5/public_html/app/Mage.php(640): Mage_Core_Model_App->run(Array)
#39 /home/niland5/public_html/index.php(85): Mage::run('', 'store')
#40 {main}
```

I believe my local.xml is standard, but I have included it here:

```
        <session_save><![CDATA[db]]></session_save>
        <redis_session>                       <!-- All options seen here are the defaults -->
            <host>127.0.0.1</host>            <!-- Specify an absolute path if using a unix socket -->
            <port>6379</port>
            <password></password>             <!-- Specify if your Redis server requires authentication -->
            <timeout>2.5</timeout>            <!-- This is the Redis connection timeout, not the locking timeout -->
            <persistent></persistent>         <!-- Specify unique string to enable persistent connections. -->
            <db>2</db>                        <!-- Redis database number; protection from accidental loss is improved by using a unique DB number for sessions -->
            <compression_threshold>2048</compression_threshold>  <!-- Set to 0 to disable compression (recommended when suhosin.session.encrypt=on); known bug with strings over 64k: https://github.com/colinmollenhour/Cm_Cache_Backend_Redis/issues/18 -->
            <compression_lib>gzip</compression_lib>              <!-- gzip, lzf or snappy -->
            <log_level>4</log_level>               <!-- 0 (emergency: system is unusable), 4 (warning; additional information, recommended), 5 (notice: normal but significant condition), 6 (info: informational messages), 7 (debug: the most information for development/testing) -->
            <max_concurrency>6</max_concurrency>                 <!-- maximum number of processes that can wait for a lock on one session; for large production clusters, set this to at least 10% of the number of PHP processes -->
            <break_after_frontend>5</break_after_frontend>       <!-- seconds to wait for a session lock in the frontend; not as critical as admin -->
            <break_after_adminhtml>30</break_after_adminhtml>
            <bot_lifetime>7200</bot_lifetime>                    <!-- Bots get shorter session lifetimes. 0 to disable -->
        </redis_session>
<cache>
  <options>
   <backend>Mage_Cache_Backend_Redis</backend> <!--  Cm_Cache_Backend_Redis -->
    <backend_options>
      <server>127.0.01</server> <!-- or absolute path to unix socket -->
      <port>6379</port>
      <persistent></persistent> <!-- Specify unique string to enable persistent connections -->
      <database>0</database> <!-- Redis database number; protection against accidental data loss is improved by not sharing databases -->
      <password></password> <!-- Specify if your Redis server requires authentication -->
      <force_standalone>0</force_standalone>  <!-- 0 for phpredis, 1 for standalone PHP -->
      <connect_retries>1</connect_retries>    <!-- Reduces errors due to random connection failures; value=1 will not retry after the first failure -->
      <automatic_cleaning_factor>0</automatic_cleaning_factor> <!-- Disabled by default -->
      <compress_data>1</compress_data>  <!-- 0-9 for compression level, recommended: 0 or 1 -->
      <compress_tags>1</compress_tags>  <!-- 0-9 for compression level, recommended: 0 or 1 -->
      <compress_threshold>20480</compress_threshold>  <!-- Strings below this size will not be compressed -->
      <compression_lib>lzf</compression_lib> <!-- Supports gzip, lzf, lz4 (as l4z) and snappy -->
    </backend_options>
   </options>
</cache>
<cache>
 <amfpc>
  <options>
   <backend>Mage_Cache_Backend_Redis</backend> <!--  Cm_Cache_Backend_Redis -->
    <backend_options>
      <server>127.0.01</server> <!-- or absolute path to unix socket -->
      <port>6379</port>
      <persistent></persistent> <!-- Specify unique string to enable persistent connections -->
      <database>1</database> <!-- Redis database number; protection against accidental data loss is improved by not sharing databases -->
      <password></password> <!-- Specify if your Redis server requires authentication -->
      <force_standalone>0</force_standalone>  <!-- 0 for phpredis, 1 for standalone PHP -->
      <connect_retries>1</connect_retries>    <!-- Reduces errors due to random connection failures; value=1 will not retry after the first failure -->
      <automatic_cleaning_factor>0</automatic_cleaning_factor> <!-- Disabled by default -->
      <compress_data>1</compress_data>  <!-- 0-9 for compression level, recommended: 0 or 1 -->
      <compress_tags>1</compress_tags>  <!-- 0-9 for compression level, recommended: 0 or 1 -->
      <compress_threshold>20480</compress_threshold>  <!-- Strings below this size will not be compressed -->
      <compression_lib>gzip</compression_lib> <!-- Supports gzip, lzf, lz4 (as l4z) and snappy -->
    </backend_options>
   </options>
 </amfpc>
</cache>
    </global>
```

My server uses solid state drives
Thanks James Niland
nilandsplace.com
nilandsplace@gmail.com

**Comment by steverobbins** [comment]
@nilands55s What version of Magento are you using?

**Comment by steverobbins** [comment]
I'm not sure why exactly your installation is having trouble converting the timestamp to a date, but I've suppressed this warning in the latest version.

**Comment by nilands55s** [comment]
Hello
Sorry I missed your first email. I forgot I put my gmail on this blasted
windows compuert. I keep the windows just for testing. Anyway I am using
Magento 1.6.2.0 Live, but I am testing 1.9.something on another URL (Also
liver but unpublished).
Thanks James Niland

Sincerely, James Niland
*Niland Enterprise LLC*
http://nilandsplace.com

**Comment by nilands55s** [comment]
Hello Mr Robbins
I did the fix and seems to work. I will do the update later. I have another question if you would be so kind to answer? Should /var/cache be empty? it seems to have the same 16 files that are there if I were using files for cache. I know with APC it would be empty. Is this normal or is something wrong? your manager indicates is has keys stored.
I also use Amasty Full Page Cache, but they say it works with Redis. I just had to add <amfpc> right after <cache> and the </amfpc> respectively. With out digging up a bunch of websites I can't remember why in the above local.xml I have <database>0</database> and <database>1</database>, with the <amfpc> on database 1? So I don't know if there is a problem here? as the manager only indicates Name: Session 	Database: 0 	Keys: 2
It looked like session and cache where on the same database so I set the session to 0 and then cache to 1 and cache amfpc to 3. Would this be right? my understanding of all of this is minimal as my mother now has cancer now and much of my time is spent with her care so I have not had time to study what I am doing. Any help would be great?
Thanks James Niland
[local.txt](https://github.com/steverobbins/Magento-Redismanager/files/133165/local.txt)
![redis-manager_2016-02-16 14 38 26](https://cloud.githubusercontent.com/assets/8949704/13088702/57593d40-d4bb-11e5-814e-9258be0ace8f.png)
